### PR TITLE
restructuring Metrics into 101, 201, 301 level guides

### PR DESCRIPTION
**Pull request by asibyl**

### CODE REVIEW

File: docs/metrics/create-metric-tags.md [hunk 1]
```diff
@@ -1,11 +1,9 @@
 ---
 title: Creating Metric Tags
 sidebar_label: Creating Metric Tags
-slug: /metrics/create-metric-tags
+slug: /metrics/metrics-201/create-metric-tags
 ---
```
Comment (tore-statsig) [inline docs/metrics/create-metric-tags.md:5]: I thought we were going to leave the urls alone so links didnt break, but just rename them inline and in the sidebar

I'm okay with going this route, long term its not a concern and doing the change sooner is better

But while we are at it:
/metrics/metrics- is a bit redundant, how about just /metrics/201/create-metric-tags?

File: docs/metrics/metrics-301.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+title: Metrics 301
+sidebar_label: Metrics 301
+slug: /metrics/metrics-301
+---
+
+# Metrics 301
```
Comment (tore-statsig) [inline docs/metrics/metrics-301.md:7]: what goes in 301?

File: sidebars.js [hunk 3]
```diff
@@ -140,12 +140,18 @@ module.exports = {
...
       },
       items: [
+        "metrics/metrics-101",
+        "metrics/raw-events",
+        "metrics/raw-event-metrics",
+        "metrics/custom-metrics",
+        "metrics/precomputed-metrics",
+        "metrics/pulse-metrics",
+        "metrics/metric-dimensions",
+        "metrics/metrics-201",
```
Comment (tore-statsig) [inline sidebars.js:150]: I can help turn these into folders

**Review by tore-statsig — COMMENTED** [review]
Build is failing for broken links.  If you click the red X, you can see here:

```
Error:  Error: Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /metrics:
   -> linking to metrics-101 (resolved as: /metrics-101)
   -> linking to metrics-[20](https://github.com/statsig-io/docs/runs/7621794632?check_suite_focus=true#step:5:21)1 (resolved as: /metrics-201)
   -> linking to metrics-301 (resolved as: /metrics-301)

- On source page path = /metrics/metrics-101/custom-metrics:
   -> linking to /metrics/create

- On source page path = /metrics/metrics-101/metrics-from-events:
   -> linking to metrics/user (resolved as: /metrics/metrics-101/metrics/user)
```

File: docs/metrics/metrics-101.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+title: Metrics 101
+sidebar_label: Metrics 101
+slug: /metrics/metrics-101
```
Comment (tore-statsig) [inline docs/metrics/metrics-101.md:4]: metrics/101 should be a folder full of all the 101 content

**Review by tore-statsig — COMMENTED** [review]
In the `sidebars.js` file, a folder under featuregates/implement looks like this:

```
items: [
        "feature-gates/create-new",
        "feature-gates/add-rule",
        "feature-gates/test-gate",
        "feature-gates/overrides",
        {
          Implement: [
            "feature-gates/implement",
            "feature-gates/implement/client",
            "feature-gates/implement/server",
            "feature-gates/implement/http-api",
          ],
        },
        "feature-gates/conditions",
        "feature-gates/view-exposures",
        "feature-gates/best-practices",
      ],
    },
    ```

**Comment by jasonwzm** [comment]
🚀 Deployed on https://62e984ab413e715eb0179dc3--cozy-fox-0defec.netlify.app

**Pull request merged**